source/game_sa/CStats.cpp

```diff
@@ -36,93 +35,170 @@ unsigned int& CStats::m_AddToHealthCounter = *(unsigned int*)0xB79510;
+    ReversibleHooks::Install("CStats", "LoadActionReactionStats", 0x5599B0, &CStats::LoadActionReactionStats);
+    ReversibleHooks::Install("CStats", "FindMaxNumberOfGroupMembers", 0x559A50, &CStats::FindMaxNumberOfGroupMembers);
+    ReversibleHooks::Install("CStats", "ProcessReactionStatsOnDecrement", 0x559730, &CStats::ProcessReactionStatsOnDecrement);
+    ReversibleHooks::Install("CStats", "DecrementStat", 0x559FA0, &CStats::DecrementStat);
+    ReversibleHooks::Install("CStats", "SetNewRecordStat", 0x55C410, &CStats::SetNewRecordStat);
+    ReversibleHooks::Install("CStats", "RegisterFastestTime", 0x55A0B0, &CStats::RegisterFastestTime);
+    ReversibleHooks::Install("CStats", "RegisterBestPosition", 0x55A160, &CStats::RegisterBestPosition);
     ReversibleHooks::Install("CStats", "ProcessReactionStatsOnIncrement", 0x55B900, &CStats::ProcessReactionStatsOnIncrement);
+    ReversibleHooks::Install("CStats", "IncrementStat", 0x55C180, &CStats::IncrementStat);
+    ReversibleHooks::Install("CStats", "UpdateStatsWhenSprinting", 0x55C660, &CStats::UpdateStatsWhenSprinting);
     ReversibleHooks::Install("CStats", "UpdateStatsWhenRunning", 0x55C6F0, &CStats::UpdateStatsWhenRunning);
     ReversibleHooks::Install("CStats", "UpdateStatsWhenOnMotorBike", 0x55CD60, &CStats::UpdateStatsWhenOnMotorBike);
     ReversibleHooks::Install("CStats", "UpdateStatsWhenFighting", 0x55CFA0, &CStats::UpdateStatsWhenFighting);
-    ReversibleHooks::Install("CStats", "SafeToShowThisStat", 0x559590, &CStats::SafeToShowThisStat);
-}
+    ReversibleHooks::Install("CStats", "ModifyStat", 0x55D090, &CStats::ModifyStat);
 
-// Unused
-char* CStats::GetStatID(eStats stat) {
-    if (stat >= 0x52u)
-        sprintf(gString, "stat_i_%d", stat);
-    else
-        sprintf(gString, "stat_f_%d", stat);
+    // unused
+    ReversibleHooks::Install("CStats", "GetStatID", 0x558DE0, &CStats::GetStatID);
+    ReversibleHooks::Install("CStats", "GetTimesMissionAttempted", 0x558E70, &CStats::GetTimesMissionAttempted);
+    ReversibleHooks::Install("CStats", "RegisterMissionAttempted", 0x558E80, &CStats::RegisterMissionAttempted);
+    ReversibleHooks::Install("CStats", "RegisterMissionPassed", 0x558EA0, &CStats::RegisterMissionPassed);
 
-    return gString;
+    ReversibleHooks::Install("CStats", "Save", 0x5D3B40, &CStats::Save);
+    ReversibleHooks::Install("CStats", "Load", 0x5D3BF0, &CStats::Load);
 }
 
-// Unused
-bool CStats::GetStatType(eStats stat) {
-    return plugin::CallAndReturn<bool, 0x558E30, eStats>(stat);
-    //return (unsigned __int16)this < 0x52u;
+// 0x55C0C0
+void CStats::Init() {
+    memset(StatTypesFloat, 0, sizeof(StatTypesFloat));
+    memset(StatTypesInt, 0, sizeof(StatTypesInt));
+    memset(PedsKilledOfThisType, 0, sizeof(PedsKilledOfThisType));
+    memset(TimesMissionAttempted, 0, sizeof(TimesMissionAttempted));
+
+    bStatUpdateMessageDisplayed = false;
+    CTimer::m_snTimeInMilliseconds = 0;
+    LastMissionPassedName[0] = NULL;
```

Comment:
```suggestion
    LastMissionPassedName[0] = '\0';
    for (int32_t i = 0; i < 8; i++) {
        LastMissionPassedName[i] = '\0';
    }
```
![image](https://user-images.githubusercontent.com/19859127/128831754-24522a9a-171a-485f-84f5-2c194f4c2eac.png)

You need to set all 8 bytes to null. Since this is a string, it's better to use the null terminator `'\0'`